apps/core/views/emails.py

```diff
@@ -64,17 +65,21 @@ def get(self, request, *args, **kwargs):
 
             return Response("User not found", status=status.HTTP_404_NOT_FOUND)
 
-        if user.is_email_verified:
-            return Response("Email is already verified", status=status.HTTP_400_BAD_REQUEST)
-
         if not default_token_generator.check_token(user, confirmation_token):
             return Response(
                 "Token is invalid or expired. "
                 "Please request another confirmation email by signing in.",
                 status=status.HTTP_400_BAD_REQUEST,
             )
 
-        user.is_email_verified = True
+        if user.is_email_verified and user.email == email:
+            return Response("Email is already verified", status=status.HTTP_400_BAD_REQUEST)
+
+        if user.is_email_verified and user.email != email:
+            user.email = email
+        else:
+            user.is_email_verified = True
+
```

Comment:
Looks like readability can be improved here, maybe chatgpt refactoring or something like it can suggest ways of making it easier to understand